cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,348 @@
+
+const (
+	credentials         = "credentials"
+	regPullCredsAll     = "registry-pull-credentials-all"
+	dotDockerConfigJson = ".dockerconfigjson"
+	arm01               = "arm01"
+	testCredentials     = "test-credentials"
+	kubeconfig          = "kubeconfig"
+	config              = "config"
+)
+
+func updateCiSecretBootstrap(o options) error {
+	secretBootstrapDir := filepath.Join(o.releaseRepo, "core-services", "ci-secret-bootstrap")
+	secretBootstrapConfigFile := filepath.Join(secretBootstrapDir, "_config.yaml")
+	logrus.Infof("Updating ci-secret-bootstrap: %s", secretBootstrapConfigFile)
+
+	var c secretbootstrap.Config
+	if err := secretbootstrap.LoadConfigFromFile(secretBootstrapConfigFile, &c); err != nil {
+		return err
+	}
+	if err := updateCiSecretBootstrapConfig(o, &c); err != nil {
+		return err
+	}
+	return secretbootstrap.SaveConfigToFile(secretBootstrapConfigFile, &c)
+}
+
+func updateCiSecretBootstrapConfig(o options, c *secretbootstrap.Config) error {
+	for _, groupName := range []string{buildUFarm, "non_app_ci", "non_app_ci_x86"} {
+		c.ClusterGroups[groupName] = append(c.ClusterGroups[groupName], o.clusterName)
+	}
+	c.UserSecretsTargetClusters = append(c.UserSecretsTargetClusters, o.clusterName)
+	if err := updatePodScalerSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateBuildFarmSecrets(c, o); err != nil {
+		return err
+	}
+	if err := updateDPTPControllerManagerSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateRehearseSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateChatBotSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateExistingRegistryPullCredentialsAllSecrets(c, o); err != nil {
+		return err
+	}
+	appendSecret(generateRegistryPushCredentialsSecret, c, o)
+	appendSecret(generateRegistryPullCredentialsSecret, c, o)
+	appendSecret(generateCiOperatorSecret, c, o)
+	generateRegistryPullCredentialsAllSecrets(c, o)
+	return nil
+}
+
+func appendSecret(secretGenerator func(options) secretbootstrap.SecretConfig, c *secretbootstrap.Config, o options) {
+	secret := secretGenerator(o)
+	logrus.Infof("Creating new secret with 'to' of: %v", secret.To)
+	c.Secrets = append(c.Secrets, secret)
+}
+
+func generateCiOperatorSecret(o options) secretbootstrap.SecretConfig {
+	return secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			kubeconfig: {
+				Field: serviceAccountKubeconfigPath(ciOperator, o.clusterName),
+				Item:  buildUFarm,
+			},
+		},
+		To: []secretbootstrap.SecretContext{
+			{
+				Cluster:   o.clusterName,
+				Name:      ciOperator,
+				Namespace: testCredentials,
+			},
+		},
+	}
+}
+
+func generateRegistryPushCredentialsSecret(o options) secretbootstrap.SecretConfig {
+	return secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			dotDockerConfigJson: generatePushPullSecretFrom(o.clusterName, []secretbootstrap.DockerConfigJSONData{
+				{
+					AuthField:   "token_image-pusher_app.ci_reg_auth_value.txt",
```

Comment:
Don't you have a helper for this already?

```go
func registryAuthFor(serviceAccount, cluster string) string {}
```

Reply:
I have a `registryUrlFor` function, but I'm not seeing any existing function that would help out here. I could create one, but I believe this would be the sole usage. Thoughts?

Reply:
Is it not `registryPullTokenField`?

Reply:
ah...this one is a `pusher` though. I could modify that method to do both.